Tasks/ConsoleApp/AsyncAwait/ProductsContext.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.EntityFrameworkCore;
+namespace AsyncAwait
+{
+    public class ProductsContext : DbContext
+    {
+        public DbSet<Product> _products { get; set; }
+
+        private string _connectionString { get; set; }
+
+        public ProductsContext(string connectionString = @"Server=(localdb)\mssqllocaldb;Database=Blogging;Integrated Security=True")
+        {
+            Database.EnsureCreated();
+            _connectionString = connectionString;
+        }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            optionsBuilder.UseSqlServer(@"Server=(localdb)\mssqllocaldb;Database=Blogging;Integrated Security=True");
```

Comment:
So this code is still here. Moving connection string is the move in right direction, but this class still decides how to configure itself. What if I want to another database, mysql for example, or SQLite? You are hardcoding that we are always will use SqlServer.